main.js

```diff
@@ -430,21 +426,24 @@ async function createWindow() {
 
     // On Mac, or on other platforms when the tray icon is in use, the window
     // should be only hidden, not closed, when the user clicks the close button
-    if (
-      !windowState.shouldQuit() &&
-      (usingTrayIcon || process.platform === 'darwin')
-    ) {
-      // toggle the visibility of the show/hide tray icon menu entries
-      if (tray) {
-        tray.updateContextMenu();
+    if (!windowState.shouldQuit()) {
+      let stayInBackground = userConfig.get('stayInBackground');
+
+      if (!_.isBoolean(stayInBackground)) {
+        showStayInBackgroundWindow();
+
+        userConfig.set('stayInBackground', true);
+        stayInBackground = true;
```

Comment:
I would have expected the stayInBackgroundWindow to be a prompt asking if the user would like the app to keep running in the background, perhaps with a message that they won't get notifications if the app is completely closed, and stayInBackground be set based on the user response.

Reply:
Yeah, that was my first thought too, but after thinking a bit about this i had two conclusions:

1. Users dont want prompts, they expect the default behaviour to be the appropriate one.
2. I never saw an app that was actually asking the question `You clicked on the close button, would you like the app to actually close, or keep running in the background ?` 

At the end this really depends on who the application is targeting. If the target audience is the general public, then i believe the actual behaviour is correct.

On the other hand, if `Signal` targets specifically  '**security aware**' users or **'power users'**, then having a prompt that explicitly ask for user acknowledgment would be a better solution.

If there is a general consensus that a prompt should be used, just provide the text/wording you want in the dialog and I'll make the changes.

Reply:
Actually, on Linux I've seen several applications ask on the very first close whether the user wants to minimise to tray or terminate the application. I'm not sure if it's abnormal behaviour on other platforms though.

Reply:
I think you can't go wrong with that, I've also seen this one, actually Microsofts own Teams for example doesn't ask per se, but at least shows a notification saying it's staying in the background giving the user an option to turn it off, so something like a dialog can't be bad.